### PR TITLE
Docs: change release-notes to the changelog

### PR DESCRIPTION
**Pull request by Skoczowianin**
### Context

Changelogs are more suitable than release notes for developers seeking the details of specific change

### How has this been tested?
Locally, Staging

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/dev-handsontable/issues/1956
### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

File: docs/docker/redirects.conf [hunk 1]
```diff
@@ -32,6 +32,9 @@ if ($cookie_docs_fw = "react") {
 rewrite ^/docs/?$ /docs/$framework/ permanent;
 rewrite ^/docs/(\d+\.\d+|next)/?$ /docs/$1/$framework/ permanent;
 
+# --- redirect /release-notes/ to /changelog/ ---
+rewrite ^/docs/$framework/release-notes/?$ /docs/$framework/changelog/ permanent;
```
Comment (budnix) [inline docs/docker/redirects.conf:36]: I'd move this line to line 124 and change it to
```
rewrite ^/docs/(javascript|react)-data-grid/release-notes/?$ /docs/$framework/changelog/ permanent;
```

**Review by budnix — CHANGES_REQUESTED** [review]
Some changes need correction. Also, the link "Changelog" in the Docs header does not work (404). Can you check that? 
Zrzut ekranu 2024-07-10 o 10 25 16

**Review by budnix — APPROVED** [review]
👍

**Comment by magierg** [comment]
fix verified on staging

**Pull request merged**